src/lib/eap_aka_sim/state_machine.c

```diff
@@ -169,15 +168,17 @@ static bool identity_req_set_by_user(request_t *request, eap_aka_sim_session_t *
 	 *	then delete them so they don't screw
 	 *	up any of the other code.
 	 */
-	for (vp = fr_cursor_init(&cursor, &request->reply_pairs);
+	for (vp = fr_pair_list_head(&request->reply_pairs);
 	     vp;
-	     vp = fr_cursor_next(&cursor)) {
+	     vp = fr_pair_list_next(&request->reply_pairs, vp)) {
 		if (vp->da == attr_eap_aka_sim_permanent_id_req) {
 			eap_aka_sim_session->id_req = AKA_SIM_PERMANENT_ID_REQ;
 		found:
 			set_by_user = true;
 			RDEBUG2("Previous section added &reply.%pP, will request additional identity", vp);
-			fr_cursor_free_item(&cursor);
+			prev = fr_pair_list_prev(&request->reply_pairs, vp);
+			fr_pair_delete(&request->reply_pairs, vp);
```

Comment:
`fr_pair_delete` vs `fr_pair_remove`? Does one free the pair? Maybe have fr_pair_delete return the previous pair to make iteration easier

Reply:
fr_pair_remove() just removes the pair from the list (mainly used where pairs are being moved from one list to another), fr_pair_delete() also frees.  I'll add the return of the previous pair.